AmazonLinux 2 build errors

CMake doesn't know how to download missing dependencies. WIP on a dependency installer so it builds as expected under CMake for Fedora/Redhat/AmazonLinux distros.


```bash
mkdir build; cd build; cmake ../

-- The CXX compiler identification is GNU 7.3.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Downloading conan.cmake from https://github.com/conan-io/cmake-conan
-- Conan: checking conan executable
CMake Error at build/conan.cmake:531 (message):
  Conan executable not found!
Call Stack (most recent call first):
  build/conan.cmake:565 (conan_check)
  src/CMakeLists.txt:28 (conan_add_remote)
```

```bash
#pip3 install conan; mkdir build; cd build; cmake ../

-- The CXX compiler identification is GNU 7.3.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Downloading conan.cmake from https://github.com/conan-io/cmake-conan
-- Conan: checking conan executable
-- Conan: Found program /usr/local/bin/conan
-- Conan: Version found Conan version 1.34.0

-- Conan: Adding bincrafters remote repository (https://api.bintray.com/conan/bincrafters/public-conan)
WARN: Remotes registry file missing, creating default one in /root/.conan/remotes.json
-- Conan: Automatic detection of conan settings from cmake
CMake Error at build/conan.cmake:82 (message):
  Please specify in command line CMAKE_BUILD_TYPE
  (-DCMAKE_BUILD_TYPE=Release)
Call Stack (most recent call first):
  build/conan.cmake:496 (conan_cmake_settings)
  src/CMakeLists.txt:33 (conan_cmake_run)


-- Configuring incomplete, errors occurred!
See also "/root/github/turtlebrowser/build/CMakeFiles/CMakeOutput.log".
```

```bash
# pip3 install conan; mkdir build; cd build; cmake -DCMAKE_BUILD_TYPE=Debug ../

Downloading conan_package.tgz
fontconfig/2.13.92: Package installed 1e8fcc6b58b09849d3e76b7562f37bcf44864cdf
fontconfig/2.13.92: Downloaded package revision 0
bzip2/1.0.8: Appending PATH environment variable: /root/.conan/data/bzip2/1.0.8/_/_/package/0e2fc6376534b5b3953a4974cf0901b008b17f2f/bin
ERROR: Not able to automatically detect '/usr/bin/c++' version
ERROR: Unable to find a working compiler
WARN: libmysqlclient/8.0.17: requirement openssl/1.1.1i overridden by qt/5.15.2@bincrafters/stable to openssl/1.1.1h
egl/system: WARN: Don't know how to install EGL for your distro.
ERROR: egl/system: Error in package_info() method, line 75
        self._fill_cppinfo_from_pkgconfig('egl')
while calling '_fill_cppinfo_from_pkgconfig', line 24
        if not pkg_config.provides:
        ConanException: pkg-config command ['pkg-config', '--print-provides', 'egl', '--print-errors'] failed with error: Command 'pkg-config --print-provides egl --print-errors' returned non-zero exit status 1.
Package egl was not found in the pkg-config search path.
Perhaps you should add the directory containing `egl.pc'
to the PKG_CONFIG_PATH environment variable
No package 'egl' found

CMake Error at build/conan.cmake:402 (message):
  Conan install failed='1'
Call Stack (most recent call first):
  build/conan.cmake:497 (conan_cmake_install)
  src/CMakeLists.txt:33 (conan_cmake_run)
```

--- a/conans/client/tools/oss.py
+++ b/conans/client/tools/oss.py
@@ -204,7 +204,7 @@ class OSInfo(object):
     def with_yum(self):
         return self.is_linux and self.linux_distro in ("pidora", "fedora", "scientific", "centos",
                                                        "redhat", "rhel", "xenserver", "amazon",
-                                                       "oracle")
+                                                       "oracle", "amzn")
 
     @property
     def with_dnf(self):

